src/Uno.UI/UI/Xaml/Media/Brush.skia.cs

```diff
@@ -16,95 +16,69 @@ public abstract partial class Brush
 		/// </summary>
 		internal static IDisposable AssignAndObserveBrush(Brush brush, Action<Color> colorSetter, Action imageBrushCallback = null)
 		{
+			colorSetter(Colors.Transparent);
+
 			if (brush == null)
 			{
-				colorSetter(SolidColorBrushHelper.Transparent.Color);
-
 				return null;
 			}
 
 			var disposables = new CompositeDisposable();
+
+			void UpdateColor(object sender, DependencyPropertyChangedEventArgs e) => colorSetter(Colors.Transparent);
+			void UpdateColorWhenAnyChanged<T>(T source, params DependencyProperty[] properties) where T : DependencyObject
+			{
+				foreach (var property in properties)
+				{
+					source
+						.RegisterDisposablePropertyChangedCallback(property, UpdateColor)
+						.DisposeWith(disposables);
+				}
+			}
+
 			if (brush is SolidColorBrush colorBrush)
 			{
-				brush.RegisterDisposablePropertyChangedCallback(
+				UpdateColorWhenAnyChanged(colorBrush, new[]
```

Comment:
you don't need to explicitly initialize the array here

Reply:
True, but I would have to align the parameters without the `new[]`.